Orange/widgets/unsupervised/owdistancefile.py

```diff
@@ -0,0 +1,158 @@
+
+
+class OWDistanceFile(widget.OWWidget, RecentPathsWComboMixin):
+    name = "Distance File"
+    id = "orange.widgets.unsupervised.distancefile"
+    description = "Read distances from file"
+    icon = "icons/DistanceFile.svg"
+    priority = 10
+    category = "Data"
+    keywords = ["data", "distances", "load", "read"]
+    outputs = [("Distances", DistMatrix)]
+
+    want_main_area = False
+    resizing_enabled = False
+
+    def __init__(self):
+        super().__init__()
+        RecentPathsWComboMixin.__init__(self)
+        self.loaded_file = ""
+
+        vbox = gui.vBox(self.controlArea, "Distance File", addSpace=True)
+        box = gui.hBox(vbox)
+        self.file_combo.setMinimumWidth(300)
+        box.layout().addWidget(self.file_combo)
+        self.file_combo.activated[int].connect(self.select_file)
+
+        button = gui.button(box, self, '...', callback=self.browse_file)
+        button.setIcon(self.style().standardIcon(QtGui.QStyle.SP_DirOpenIcon))
+        button.setSizePolicy(
+            QtGui.QSizePolicy.Maximum, QtGui.QSizePolicy.Fixed)
+
+        button = gui.button(
+            box, self, "Reload", callback=self.reload, default=True)
+        button.setIcon(self.style().standardIcon(QtGui.QStyle.SP_BrowserReload))
+        button.setSizePolicy(QtGui.QSizePolicy.Fixed, QtGui.QSizePolicy.Fixed)
+
+        box = gui.vBox(self.controlArea, "Info", addSpace=True)
+        self.infoa = gui.widgetLabel(box, 'No data loaded.')
+        self.warnings = gui.widgetLabel(box, ' ')
+        #Set word wrap, so long warnings won't expand the widget
+        self.warnings.setWordWrap(True)
+        self.warnings.setSizePolicy(
+            QtGui.QSizePolicy.Ignored, QtGui.QSizePolicy.MinimumExpanding)
+
+        self.set_file_list()
+        QtCore.QTimer.singleShot(0, self.open_file)
+
+    def set_file_list(self):
+        super().set_file_list()
+        self.file_combo.addItem("Browse documentation data sets...")
+
+    def reload(self):
+        return self.open_file()
+
+    def select_file(self, n):
+        if n < len(self.recent_paths):
+            super().select_file(n)
+        elif n:
+            self.browse_file(True)
+
+        self.set_file_list()
+        self.open_file()
+
+    def browse_file(self, in_demos=0):
+        if in_demos:
+            try:
+                start_file = get_sample_datasets_dir()
+            except AttributeError:
+                start_file = ""
+            if not start_file or not os.path.exists(start_file):
+                widgets_dir = os.path.dirname(gui.__file__)
+                orange_dir = os.path.dirname(widgets_dir)
+                start_file = os.path.join(orange_dir, "doc", "datasets")
+            if not start_file or not os.path.exists(start_file):
+                d = os.getcwd()
+                if os.path.basename(d) == "canvas":
+                    d = os.path.dirname(d)
+                start_file = os.path.join(os.path.dirname(d), "doc", "datasets")
```

Comment:
OK, I also removed it from the File Widget. Please check whether I understood it correctly.